rag_agent.py

```diff
@@ -112,25 +102,50 @@ def format_docs(docs):
                 "question": RunnablePassthrough()  
             }
             | self.prompt  
+            | (lambda x: str(x))  # Convert ChatPromptValue to string
             | self.model   
+            | (lambda outputs: outputs[0]['generated_text'])  # Extract generated text
             | StrOutputParser()  
         )
 
-        while True:
-            question = input().strip()
+        doc_result, relevance_score = self.get_relevant_document(question)
+        # Classifying query based on its relevance with retrieved context
+        if doc_result:
+            response = qa_chain.invoke({"query": question, "context": doc_result.page_content})
+        else:
+            response = qa_chain.invoke(question)
+        
+        return response
 
-            doc_result, relevance_score = self.get_relevant_document(question)
-            # Classifying query based on it's relevance with retrieved context
-            if doc_result:
-                response = qa_chain.invoke({"query": question, "context": doc_result.page_content})
-            else:
-                response = qa_chain.invoke(question)
-            
-            return response
 
 
-if __name__ == "__main__":
-    agent = RAG_Agent()
-    agent.retriever = agent.setup_vectorstore(document_paths)  
-    agent.run()
+def main():
+    parser = argparse.ArgumentParser(description="Pippy's AI-Coding Assistant")
+    parser.add_argument('--model', type=str, choices=[
+        'bigscience/bloom-1b1',
+        'facebook/opt-350m',
+        'EleutherAI/gpt-neo-1.3B'
+    ], default='bigscience/bloom-1b1', help='Model name to use for text generation')
+    parser.add_argument('--docs', nargs='+', default=[
+        './docs/Pygame Documentation.pdf',
+        './docs/Python GTK+3 Documentation.pdf',
+        './docs/Sugar Toolkit Documentation.pdf'
+    ], help='List of document paths to load into the vector store')
+    args = parser.parse_args()
+
+    try:
+        agent = RAG_Agent(model=args.model)
+        agent.retriever = agent.setup_vectorstore(args.docs)
 
+        while True:
+            question = input("Enter your question: ").strip()
+            if not question:
+                print("Please enter a valid question.")
+                continue
+            response = agent.run(question)
+            print("Response:", response)
+    except Exception as e:
+        print(f"An error occurred: {e}")
```

Comment:
Considering this is an API, we definitely wouldn't use command line args for it.